recipe/patches/0002-fix-opt_dep-attribute-check.patch

```diff
@@ -0,0 +1,28 @@
+From d01d2dea1875a302e73a37a23bccf53114fd4133 Mon Sep 17 00:00:00 2001
+From: Lorenzo Pirritano <lpirritano@anaconda.com>
+Date: Tue, 25 Jun 2024 14:36:30 +0200
+Subject: [PATCH 2/2] fix opt_dep attribute check
+
+opt_dep is not there when the tests are not run in the CI
```

Comment:
You can pass `--opt-dep` to pytest. The function says:
>    Tests that use this fixture must be run with the --opt-dep option via terminal. If not they will skip. This fixture is used in CI to indicate the optional dependencies installed in the tox environments the tests are run in. See setup.cfg and .github/workflows/ci.yml for more details.

Also see https://github.com/SeldonIO/alibi-detect/blob/master/setup.cfg#L66-L127.

Reply:
Somehow using `--pyargs` made it not possible to pass `--opt-dep`, so I removed `--pyargs` and used `--opt-dep=all`. So, now we can run tests for all optional dependencies (`tensorflow`, `prophet`, `pytorch`, except `pykeops` which we do not have). Thank you, your comment helped improving the tests.